embedding/embedlite/embedshared/EmbedLiteWindowBaseChild.cpp

```diff
@@ -182,7 +190,14 @@ void EmbedLiteWindowBaseChild::CreateWidget()
               &widgetInit              // HandleWidgetEvent
               );
   GetWidget()->UpdateSize();
+
+  initialized = true;
+
   Unused << SendInitialized();
+
+  if (destroyAfterInit) {
+    RecvDestroy();
+  }
```

Comment:
This doesn't look like right order... Very likely destroying here and sending initialized can lead to bad state.

Without checking the whole init view code path down to the browser side, following should be safer:
```
if (mDestroyAfterInit) {
    RecvDestroy();
} else {
    Unused << SendInitialized();
}
```
Does it even make sense to create nsWindow if mDestroyAfterInit is true, looks suspicious? 

Right after mCreateWidgetTask clearing... I think you could do:

```
if (mDestroyAfterInit) {
    RecvDestroy();
    return;
}
```